### ISSUE TITLE
Convert UI module to Kotlin

### ISSUE THREAD

**Issue by Fabi755** [maplibre/maplibre-navigation-android#128]
Convert UI module `libandroid-navigation-ui` to Kotlin. But at minimum the classes that are interacting with the core module models, that always have a `Builder`. The builder pattern are only added to keep simple access to the Java code, and is only a temporary solution.

**Comment by boldtrn** [comment]
Yes I agree, a bit of cleaning up of the UI module wouldn't hurt either 👍
